### PR TITLE
Add a new chart: rancher-provisioning-capi

### PR DESCRIPTION
**Pull request by Oats87**
## Issue: 
https://github.com/rancher/rancher/issues/41094

## Problem
There is no chart to install cluster-api core controllers into a Rancher cluster.

## Solution
Add a cluster-api CAPI chart.

## Testing
The v2prov integration tests (and corresponding QA integration tests that use the CAPR planner) will verify functionality.

### CODE REVIEW

**Review by jakefhyde — DISMISSED** [review]

File: index.yaml [hunk 1]
```diff
@@ -13544,6 +13544,32 @@ entries:
...
+  rancher-provisioning-capi:
+  - annotations:
+      catalog.cattle.io/certified: rancher
+      catalog.cattle.io/display-name: Rancher Provisioning CAPI Controller Manager
+      catalog.cattle.io/kube-version: '>=1.23.0-0'
+      catalog.cattle.io/namespace: cattle-provisioning-capi-system
+      catalog.cattle.io/os: linux
+      catalog.cattle.io/permits-os: linux,windows
+      catalog.cattle.io/provides-gvr: apps.deployment/v1
+      catalog.cattle.io/rancher-version: '>= 2.7.0-0'
```
Comment (jiaqiluo) [inline index.yaml:13557]: 1. Should it be at least 2.7.7? 
2. Should the chart be hidden in Rancher UI? If so we need `catalog.cattle.io/hidden: "true"`

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: index.yaml (hunk 1 above)
Comment (Oats87) [reply index.yaml:13557]: No, we don't want to lock in a specific version at this point given we don't actually know the next .x version. It's fine to leave it as `2.7.0`

Good catch on the hidden. It was originally there, but dropped when addressing Arvind's previous comments. PR incoming to fix the chart on the `provisioning` repo.

Comment (Oats87) [reply index.yaml:13557]: PR to fix this is here: https://github.com/rancher/provisioning/pull/2 once this is merged I will merge-train it back into this PR.

Comment (Oats87) [reply index.yaml:13557]: Change pushed to add the hidden annotation

**Review by jiaqiluo — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Pull request merged**